src/runtime_src/xdp/profile/rt_profile_writers.cpp

```diff
@@ -80,7 +83,7 @@ namespace XCL {
     const int maxLength = 1024;
     char buf[maxLength];
     ssize_t len;
-    if ((len=readlink("/proc/self/exe", buf, maxLength-1)) != -1) {
+    if (((len=readlink("/proc/self/exe", buf, maxLength-1)) != -1) && (len < maxLength)) {
```

Comment:
This does not seem right.   What if len>=maxLength, e.g. the input is truncated, where is the buffer null terminated?

Reply:
Ok, I see.  That's a false positive from Coverity then.